src/main/scala/analysis/data_structure_analysis/LocalPhase.scala

```diff
@@ -310,52 +305,53 @@ class LocalPhase(proc: Procedure,
               visitPointerArithmeticOperation(n, lhsCell, arg, byteSize, true)
             case _ => ???
       case store @ MemoryStore(_, ind, expr, _, size, _) =>
-        val index: Expr = unwrapPaddingAndSlicing(ind)
-        assert(size % 8 == 0)
-        val byteSize = size / 8
-        val global = isGlobal(index, n, byteSize)
-        val stack = isStack(index, n)
-        val indexCell = graph.adjust(graph.accessIndexToSlice(store))
-        assert(!indexCell.node.get.flags.merged) // check index cell is a placeholder
-        val addressPointee: Cell = if (global.isDefined) {
-          graph.mergeCells(graph.find(global.get), indexCell)
-          graph.adjust(graph.find(global.get).getPointee)
-        } else if (stack.isDefined) {
-          graph.mergeCells(graph.find(stack.get), indexCell)
-          graph.adjust(graph.find(stack.get).getPointee)
-        } else {
-            index match {
-              case BinaryExpr(op, arg1: Variable, arg2) if op.equals(BVADD) =>
-                evaluateExpression(arg2, constProp(n)) match {
-                  case Some(v) =>
-                    //                    assert(varToSym(n).contains(arg1))
-                    val offset = v.value
-                    visitPointerArithmeticOperation(n, indexCell, arg1, byteSize, false, offset)
-                    visitPointerArithmeticOperation(n, Node(Some(graph)).cells(0), arg1, byteSize, true, offset)
-                  case None =>
-                    //                    assert(varToSym(n).contains(arg1))
-                    // collapse the results
-                    // visitPointerArithmeticOperation(n, DSN(Some(graph)).cells(0), arg1, byteSize, true, 0, true)
-                    unsupportedPointerArithmeticOperation(n, index, Node(Some(graph)).cells(0))
-                }
-              case arg: Variable =>
-                //                assert(varToSym(n).contains(arg))
-                visitPointerArithmeticOperation(n, indexCell, arg, byteSize, false)
-                visitPointerArithmeticOperation(n, Node(Some(graph)).cells(0), arg, byteSize, true)
-              case _ =>
-                ???
-            }
-        }
```

Comment:
The tests fail because you didn't keep the changed structure here which was important for correctly identifying the access indices, but when I've tried to keep this changed structure, then the DataStructureTests start failing and I'm not sure why that is.